src/api.cpp

```diff
@@ -4820,59 +4674,38 @@ cl_int CLVK_API_CALL clGetSupportedImageFormats(cl_context context,
 
     // Iterate over all known CL/VK format associations and report the CL
     // formats for which the Vulkan format is supported
-    for (auto const& clvkfmt : gFormatMaps) {
-        const cl_image_format& clfmt = clvkfmt.first;
-        auto const& fmt_support = clvkfmt.second;
+    for (auto mapping : get_format_maps()) {
+        auto image_channel_order = mapping.first.image_channel_order;
+        auto image_channel_data_type = mapping.first.image_channel_data_type;
+        VkComponentMapping components_sampled, components_storage;
+        image_format_support fmt_support;
+        cl_image_format clfmt = {
+            .image_channel_order = image_channel_order,
+            .image_channel_data_type = image_channel_data_type,
+        };
+        if (!cl_image_format_to_vulkan_format(clfmt, image_type, dev,
+                                              &fmt_support, &components_sampled,
+                                              &components_storage)) {
+            continue;
+        }
```

Comment:
If I remove this conversion, this PR makes no sense. The whole point of it is this conversion which allows to check for `is_bgra_format_not_supported_for_image1d_buffer`. That's why I did not want to iterate over the maps initially. But I agree having a getter is better. But I would prefer not to have access to the VkFormat there as people need to understand that the only mapping to consider is the one from the conversion function, not the map.

Reply:
Instead of the getter, we could have a function returning the number of elements in the map, and a function to get the `cl_image_format` (only) for a specified element:
```
const unsigned get_format_maps_size();
const cl_image_format get_format_from_format_maps_at(unsigned i);
```

Reply:
Forget about that, FormatMaps is a map... not possible to do it like that.

Reply:
Right, I'd lost track of the big picture :D. I'm sure we'll find good ways of refactoring this as more support for images makes its way into the code.